--- configure.old	2015-07-04 17:05:26.919745903 +0200
+++ configure	2015-07-04 17:06:08.738745154 +0200
@@ -102,10 +102,10 @@
 
 check_ncurses()
 {
-	if check_library NCURSES "" -lncursesw
+	if pkg_config NCURSES ncursesw "" -lncursesw
 	then
 		widechars=y
-	elif check_library NCURSES "" -lncurses || check_library NCURSES "" -lcurses
+	elif pkg_config NCURSES ncurses "" -lncurses || pkg_config NCURSES curses "" -lcurses
 	then
 		widechars=n
 		msg_error "Your ncurses does not support wide characters!"
